### PR TITLE
Split RichString_(append|appendn|write) into wide and ascii

### PR DESCRIPTION
**Pull request by cgzones**
RichString_writeFrom takes a top spot during performance analysis due to the
calls to mbstowcs() and iswprint().

Most of the time we know in advance that we are only going to print regular
ASCII characters.

### CODE REVIEW

File: SwapMeter.c [hunk 1]
```diff
@@ -33,12 +33,12 @@ static void SwapMeter_updateValues(Meter* this, char* buffer, int size) {
 static void SwapMeter_display(const Object* cast, RichString* out) {
    char buffer[50];
    const Meter* this = (const Meter*)cast;
-   RichString_write(out, CRT_colors[METER_TEXT], ":");
+   RichString_writeAscii(out, CRT_colors[METER_TEXT], ":");
    Meter_humanUnit(buffer, this->total, 50);
```
Comment (BenBE) [inline SwapMeter.c:37]: Pass `sizeof(buffer)` …

File: SwapMeter.c [hunk 2]
```diff
@@ -33,12 +33,12 @@ static void SwapMeter_updateValues(Meter* this, char* buffer, int size) {
...
-   RichString_append(out, CRT_colors[METER_VALUE], buffer);
+   RichString_appendAscii(out, CRT_colors[METER_VALUE], buffer);
    Meter_humanUnit(buffer, this->values[0], 50);
```
Comment (BenBE) [inline SwapMeter.c:39]: Dito.

File: zfs/ZfsArcMeter.c [hunk 3]
```diff
@@ -52,28 +52,28 @@ static void ZfsArcMeter_display(const Object* cast, RichString* out) {
    if (this->values[5] > 0) {
       char buffer[50];
       Meter_humanUnit(buffer, this->total, 50);
```
Comment (BenBE) [inline zfs/ZfsArcMeter.c:54]: Dito.

File: zfs/ZfsArcMeter.c [hunk 4]
```diff
@@ -52,28 +52,28 @@ static void ZfsArcMeter_display(const Object* cast, RichString* out) {
...
-      RichString_append(out, CRT_colors[METER_VALUE], buffer);
+      RichString_appendAscii(out, CRT_colors[METER_VALUE], buffer);
       Meter_humanUnit(buffer, this->values[5], 50);
```
Comment (BenBE) [inline zfs/ZfsArcMeter.c:56]: Dito.

File: zfs/ZfsArcMeter.c [hunk 5]
```diff
@@ -52,28 +52,28 @@ static void ZfsArcMeter_display(const Object* cast, RichString* out) {
...
-      RichString_append(out, CRT_colors[METER_TEXT], " Used:");
-      RichString_append(out, CRT_colors[METER_VALUE], buffer);
+      RichString_appendAscii(out, CRT_colors[METER_TEXT], " Used:");
+      RichString_appendAscii(out, CRT_colors[METER_VALUE], buffer);
       Meter_humanUnit(buffer, this->values[0], 50);
```
Comment (BenBE) [inline zfs/ZfsArcMeter.c:59]: Dito.

File: zfs/ZfsArcMeter.c [hunk 6]
```diff
@@ -52,28 +52,28 @@ static void ZfsArcMeter_display(const Object* cast, RichString* out) {
...
-      RichString_append(out, CRT_colors[METER_TEXT], " MFU:");
-      RichString_append(out, CRT_colors[ZFS_MFU], buffer);
+      RichString_appendAscii(out, CRT_colors[METER_TEXT], " MFU:");
+      RichString_appendAscii(out, CRT_colors[ZFS_MFU], buffer);
       Meter_humanUnit(buffer, this->values[1], 50);
```
Comment (BenBE) [inline zfs/ZfsArcMeter.c:62]: Dito.

File: zfs/ZfsArcMeter.c [hunk 7]
```diff
@@ -52,28 +52,28 @@ static void ZfsArcMeter_display(const Object* cast, RichString* out) {
...
-      RichString_append(out, CRT_colors[METER_TEXT], " MRU:");
-      RichString_append(out, CRT_colors[ZFS_MRU], buffer);
+      RichString_appendAscii(out, CRT_colors[METER_TEXT], " MRU:");
+      RichString_appendAscii(out, CRT_colors[ZFS_MRU], buffer);
       Meter_humanUnit(buffer, this->values[2], 50);
```
Comment (BenBE) [inline zfs/ZfsArcMeter.c:65]: Dito.

File: zfs/ZfsArcMeter.c [hunk 8]
```diff
@@ -52,28 +52,28 @@ static void ZfsArcMeter_display(const Object* cast, RichString* out) {
...
-      RichString_append(out, CRT_colors[METER_TEXT], " Anon:");
-      RichString_append(out, CRT_colors[ZFS_ANON], buffer);
+      RichString_appendAscii(out, CRT_colors[METER_TEXT], " Anon:");
+      RichString_appendAscii(out, CRT_colors[ZFS_ANON], buffer);
       Meter_humanUnit(buffer, this->values[3], 50);
```
Comment (BenBE) [inline zfs/ZfsArcMeter.c:68]: Dito.

File: zfs/ZfsArcMeter.c [hunk 9]
```diff
@@ -52,28 +52,28 @@ static void ZfsArcMeter_display(const Object* cast, RichString* out) {
...
-      RichString_append(out, CRT_colors[METER_TEXT], " Hdr:");
-      RichString_append(out, CRT_colors[ZFS_HEADER], buffer);
+      RichString_appendAscii(out, CRT_colors[METER_TEXT], " Hdr:");
+      RichString_appendAscii(out, CRT_colors[ZFS_HEADER], buffer);
       Meter_humanUnit(buffer, this->values[4], 50);
```
Comment (BenBE) [inline zfs/ZfsArcMeter.c:71]: Dito.

File: zfs/ZfsCompressedArcMeter.c [hunk 10]
```diff
@@ -48,17 +48,17 @@ static void ZfsCompressedArcMeter_display(const Object* cast, RichString* out) {
    if (this->values[0] > 0) {
       char buffer[50];
       Meter_humanUnit(buffer, this->total, 50);
```
Comment (BenBE) [inline zfs/ZfsCompressedArcMeter.c:50]: Dito.

File: zfs/ZfsCompressedArcMeter.c [hunk 11]
```diff
@@ -48,17 +48,17 @@ static void ZfsCompressedArcMeter_display(const Object* cast, RichString* out) {
...
-      RichString_append(out, CRT_colors[METER_VALUE], buffer);
-      RichString_append(out, CRT_colors[METER_TEXT], " Uncompressed, ");
+      RichString_appendAscii(out, CRT_colors[METER_VALUE], buffer);
+      RichString_appendAscii(out, CRT_colors[METER_TEXT], " Uncompressed, ");
       Meter_humanUnit(buffer, this->values[0], 50);
```
Comment (BenBE) [inline zfs/ZfsCompressedArcMeter.c:53]: Dito.

File: zfs/ZfsCompressedArcMeter.c [hunk 12]
```diff
@@ -48,17 +48,17 @@ static void ZfsCompressedArcMeter_display(const Object* cast, RichString* out) {
...
-      RichString_append(out, CRT_colors[METER_VALUE], buffer);
-      RichString_append(out, CRT_colors[METER_TEXT], " Compressed, ");
+      RichString_appendAscii(out, CRT_colors[METER_VALUE], buffer);
+      RichString_appendAscii(out, CRT_colors[METER_TEXT], " Compressed, ");
       ZfsCompressedArcMeter_printRatioString(this, buffer, 50);
```
Comment (BenBE) [inline zfs/ZfsCompressedArcMeter.c:56]: Dito.

File: SwapMeter.c (hunk 1 above)
Comment (cgzones) [reply SwapMeter.c]: Thanks, done

**Review by BenBE — APPROVED** [review]

**Pull request merged**